doc/sphinx/make-your-own-compositor.md

```diff
@@ -0,0 +1,248 @@
+---
+discourse: 4911,5164,5603,6756,8037
+---
+
+# Using Mir Libraries to Build Your Own Wayland Compositor
+If you've seen [the demo tutorial](explore-mir-using-demos), you're probably
+taken away by Mir's capabilities and want to use it and the supporting
+libraries to build your own Wayland compositor. This section will guide you
+through the installation and basic usage of Mir. You'll be even more impressed
+by how easy it is to create a Wayland compositor using Mir!
+
+### A barebones Mir compositor
+
+The following code block is the bare minimum you need to run a Mir compositor:
+```cpp
+#include <miral/runner.h>
+#include <miral/minimal_window_manager.h>
+#include <miral/set_window_management_policy.h>
+
+using namespace miral;
+
+int main(int argc, char const* argv[])
+{
+    MirRunner runner{argc, argv};
+
+    return runner.run_with(
+        {
+            set_window_management_policy<MinimalWindowManager>()
+        });
+}
+```
+
+You can see it's barely longer than the usual C++ "Hello, world!" example. This
+code gives you a minimal compositor with basic window management
+capabilities such as controlling multiple windows, minimizing and maximizing,
+handling input, etc... This is done with the help of MirAL (Mir Abstraction
+Layer) which gives you a high level interface to work with Mir!
+
+#### Dependencies
+To compile, you'll need to install `libmiral`, and
+`mir-graphics-drivers-desktop` which can be done on different distros as
+follows:
+
+Installing dependencies on Debian and its derivatives:
+```sh
+sudo apt install libmiral-dev mir-graphics-drivers-desktop
+```
+
+<details>
+<summary> Installing Dependencies on Other Distros </summary>
+
+Installing dependencies on Fedora 
+```sh
+sudo dnf install mir-devel libxkbcommon
+```
+Installing dependencies on Alpine 
+```sh
+sudo apk add mir-dev
+```
+</details>
```

Comment:
The tutorial can safely assume that you're on Ubuntu. This is because we can't guarantee anything about whether or not other distros are on latest Mir

Reply:
We should mention our assumptions at the top of the tutorial, e.g.:

```md
## Prerequisites
- A computer running Ubuntu 24.04
- A compiler with support for C++23
```

Reply:
Hmmm, that would make things a lot easier to test (manually and automatically) and should allow us to ditch the collapsing sections needed for other distros. @AlanGriffiths What are your thoughts on this?